src/lib/common/macroSubstitute.cpp

```diff
@@ -121,8 +121,16 @@ static void attributeValue(std::string* valueP, const std::vector<ContextAttribu
 *
 */
 #define CHUNK_SIZE 1024
-void macroSubstitute(std::string* to, const std::string& from, const ContextElement& ce)
+bool macroSubstitute(std::string* to, const std::string& from, const ContextElement& ce)
 {
+  // Initial size check: is the string to convert too big?
```

Comment:
The initial check is not 100% necessary, but imagine a buffer of 16MB enters the function. The algorithm starts and copies and copies, only to finally after copying 8MB decide to return an error.
This initial check avoids that, which was the entire reason for the 8MB limit in macroSubstitute()